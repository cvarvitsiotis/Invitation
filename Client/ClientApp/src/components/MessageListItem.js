import React from 'react';
import PropTypes from 'prop-types';

const dateDisplay = dateStr => new Date(dateStr).toDateString();

class MessageListItem extends React.PureComponent {
  render() {
    const { message } = this.props;
    return (
      <div  className="list-group-item d-flex justify-content-between">
        <div>{message.content}</div>
        <div>{dateDisplay(message.date)}</div>
      </div>
    );
  }
}

MessageListItem.propTypes = {
  message: PropTypes.shape({
    content: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired
  })
};

export default MessageListItem;