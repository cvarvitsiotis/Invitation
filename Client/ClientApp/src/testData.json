{
  "data": {
    "events": [
      {
        "id": "1",
        "description": "Betty's wedding",
        "date": "3/4/18",
        "messages": [
          {
            "id": "1",
            "content": "Hey all. Betty is having a wedding.",
            "date": "1/5/18"
          },
          {
            "id": "2",
            "content": "Don't forget about Betty's wedding.",
            "date": "1/22/18"
          }
        ],
        "personStatuses": [
          {
            "id": "1",
            "personId": "1",
            "status": "yes"
          },
          {
            "id": "2",
            "personId": "2",
            "status": "no response"
          },
          {
            "id": "3",
            "personId": "3",
            "status": "maybe"
          }
        ]
      },
      {
        "id": "2",
        "description": "Lenny's birthday party",
        "date": "4/18/18",
        "messages": [
          {
            "id": "3",
            "content": "Come to Lenny's 21st birthday party.",
            "date": "2/9/18"
          },
          {
            "id": "4",
            "content": "We have moved locations of Lenny's party.",
            "date": "2/28/18"
          }
        ],
        "personStatuses": [
          {
            "id": "4",
            "personId": "4",
            "status": "yes"
          },
          {
            "id": "5",
            "personId": "5",
            "status": "no"
          },
          {
            "id": "6",
            "personId": "6",
            "status": "not prompted"
          }
        ]
      }
    ],
    "people": [
      {
        "id": "1",
        "firstName": "Bob",
        "lastName": "Pendergast",
        "phone": "1111111111"
      },
      {
        "id": "2",
        "firstName": "Sue",
        "lastName": "Finklestein",
        "phone": "2222222222"
      },
      {
        "id": "3",
        "firstName": "Jim",
        "lastName": "Forrestor",
        "phone": "3333333333"
      },
      {
        "id": "4",
        "firstName": "Craig",
        "lastName": "Pattington",
        "phone": "4444444444"
      },
      {
        "id": "5",
        "firstName": "Olly",
        "lastName": "Fowler",
        "phone": "5555555555"
      },
      {
        "id": "6",
        "firstName": "Phil",
        "lastName": "Pumpernickel",
        "phone": "6666666666"
      }
    ]
  }
}