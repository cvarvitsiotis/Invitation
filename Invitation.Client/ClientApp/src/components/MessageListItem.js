import React from 'react';
import PropTypes from 'prop-types';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

const dateDisplay = dateStr => new Date(dateStr).toDateString();

class MessageListItem extends React.PureComponent {
  render() {
    const { message } = this.props;
    return (
      <ListItem>
        <ListItemText
          primary={message.content}
          secondary={dateDisplay(message.date)}
        />
      </ListItem>
    );
  }
}

MessageListItem.propTypes = {
  message: PropTypes.shape({
    content: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired
  })
};

export default MessageListItem;