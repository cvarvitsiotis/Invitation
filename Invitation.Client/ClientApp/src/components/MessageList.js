import React from 'react';
import MessageListItem from './MessageListItem';
import { Link } from 'react-router-dom';
import { CardHeader, CardContent } from 'material-ui/Card';
import Button from 'material-ui/Button';
import AddCircleIcon from 'material-ui-icons/AddCircle';
import List from 'material-ui/List';
import CardWithStyle from './overrides/CardWithStyle';
import { withStyles } from 'material-ui/styles';
import deepPurple from 'material-ui/colors/deepPurple';

const styles = () => ({
  button: {
    color: deepPurple[400],
    '&:hover': {
      color: deepPurple[700]
    }
  }
});

class MessageList extends React.PureComponent {
  render() {
    const { messages, match, classes } = this.props;
    return (
      <CardWithStyle>
        <CardHeader
          action={
            <Button className={classes.button} component={Link} to={`${match.url}/addMessage`}>
              <AddCircleIcon />
            </Button>
          }
          title="Messages"
        />
        <CardContent>
          <List>
            {Object.values(messages).map(message =>
              <MessageListItem
                key={message.id}
                message={message}
              />
            )}
          </List>
        </CardContent>
      </CardWithStyle>
    );
  }
}

export default withStyles(styles)(MessageList);