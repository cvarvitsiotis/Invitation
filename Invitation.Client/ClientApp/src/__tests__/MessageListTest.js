import React from 'react';
import MessageList from '../components/MessageList';
import MessageListItem from '../components/MessageListItem';
import StateApi from '../state-api/src/index';
import { data } from '../testData';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

const mountWithRouter = node => mount(<MemoryRouter>{node}</MemoryRouter>);

const store = new StateApi();
store.mapEventsAndPeopleAndTheirPropsIntoObjectsAndMerge(data);

describe('MessageList', () => {

  const messages = store.getState().events[1].messages;

  it('renders correctly', () => {
    const wrapper = mountWithRouter(
      <MessageList
        messages={messages}
        match={{'url': null}}
      />
    );

    expect(wrapper.find(MessageListItem).length).toBe(2);
    expect(wrapper.html()).toMatchSnapshot();
  });
});