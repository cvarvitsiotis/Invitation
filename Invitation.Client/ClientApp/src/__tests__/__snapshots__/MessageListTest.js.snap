// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList renders correctly 1`] = `"<div class=\\"MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 MuiCard-root-3 CardWithStyle-card-2\\"><div class=\\"MuiCardHeader-root-31\\"><div class=\\"MuiCardHeader-content-34\\"><span class=\\"MuiTypography-root-37 MuiTypography-headline-42 MuiCardHeader-title-35\\">Messages</span></div><div class=\\"MuiCardHeader-action-33\\"><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-76 MuiButton-root-61 MessageList-button-1\\" role=\\"button\\" href=\\"null/addMessage\\"><span class=\\"MuiButton-label-62\\"><svg class=\\"MuiSvgIcon-root-78\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 12H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root-84\\"></span></a></div></div><div class=\\"MuiCardContent-root-91\\"><ul class=\\"MuiList-root-92 MuiList-padding-93\\"><li class=\\"MuiListItem-root-96 MuiListItem-default-99 MuiListItem-gutters-103\\"><div class=\\"MuiListItemText-root-106\\"><h3 class=\\"MuiTypography-root-37 MuiTypography-subheading-44 MuiListItemText-primary-109\\">Hey all. Betty is having a wedding.</h3><p class=\\"MuiTypography-root-37 MuiTypography-body1-46 MuiTypography-colorTextSecondary-59 MuiListItemText-secondary-110\\">Fri Jan 05 2018</p></div></li><li class=\\"MuiListItem-root-96 MuiListItem-default-99 MuiListItem-gutters-103\\"><div class=\\"MuiListItemText-root-106\\"><h3 class=\\"MuiTypography-root-37 MuiTypography-subheading-44 MuiListItemText-primary-109\\">Don't forget about Betty's wedding.</h3><p class=\\"MuiTypography-root-37 MuiTypography-body1-46 MuiTypography-colorTextSecondary-59 MuiListItemText-secondary-110\\">Mon Jan 22 2018</p></div></li></ul></div></div>"`;
